File: problems/stocks/solutions/stocks.js
const getMaxProfit = input => {
    const stocks = input.split(' ').map(price => parseInt(price, 10));

    let maxProfit = stocks[1] - stocks[0];
    let currentMin = stocks[0];

    for (let i = 1; i < stocks.length; i++) {
        maxProfit = Math.max(maxProfit, stocks[i] - currentMin);
        currentMin = Math.min(currentMin, stocks[i]);
    }

    return maxProfit;
}



const chunks = [];
process.stdin.on('data', d => chunks.push(d));

process.stdin.on('end', () => {
    const data = chunks.join('').trim().split('\n');

    for (let i = 1; i < data.length; i++) {
        console.log(getMaxProfit(data[i]));
    }
});

File: problems/stocks/solutions/stocks.py
def get_max_profit(stocks):
    max_profit = stocks[1] - stocks[0]
    current_min = stocks[0]

    for stock in stocks[1:]:
        max_profit = max(max_profit, stock - current_min)
        current_min = min(current_min, stock)

    return max_profit

tests = int(input())

for _ in range(tests):
    stocks = [int(x) for x in input().split()]

    print(get_max_profit(stocks))